### PR TITLE
Cilium: encrypt-node fixes for invalid route and missing rp_filter zero

### PR DESCRIPTION
**Pull request by jrfastab**
When encrypt-node is enabled a route is added in the default routing table that routes the IP assigned to the encrypt-interface to cilium_host. However, this is problematic because it results in a routing loop when trying to send traffic (tested with ping) to the node.

After some amount of debugging and staring at the code it appears this happens when a route.Route is specified with a Nexthop but without a Local IPNet. As shown here,

       route.Route{
               Nexthop: nexthop,
               Device:  dev,
               Prefix:  *ip,
               Table:   tbl,
        }

Then the Nexthop address is used as the route destination for the routing table entry and the Prefix ip is ignored(?!). I verified this by putting a log statement just above the route add to ensure it was in-fact using the correct IPs. Further, I found the line that was adding this route by commenting it out and verifying the rule is in fact not added. The offending route is from reateNodeExternalIPSecOutRoute() in the dflt=true case.

There are a few workarounds for this problem. Simply not specifying a Nexthop is the easiest and is the approach we take in this PR because the Nexthop is not needed for encrypt-node and can be removed. An alternative fix is to add a Local IP field.

At this point I don't have a root-cause for the above. The above route.Route is pre-processed by ./pkg/datapath/linux/route/route_linux.go and then translated into a netlink route by the netlink/route_linux.go code in the vendor API. After this it is pushed into the kernel. Presumably, somewhere here either the translation is broken (I'm not seeing how though) or the kernel is buggy. These tests were run on GKEs ubuntu 4.15 kernel. We can root-cause later the above fix resolves this in the meantime. Further its a nice optimization even if there was no issue with the initial route.Route spec.

Lastly, this PR automatically disables rp_filter on the network facing interface. Otherwise if this defaults to '1' node to node traffic will fail.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8257)

### CODE REVIEW

**Comment by jrfastab** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]

File: pkg/datapath/linux/datapath.go [hunk 1]
```diff
@@ -43,6 +44,12 @@ func NewDatapath(config DatapathConfiguration) datapath.Datapath {
 
 	dp.node = NewNodeHandler(config, dp.nodeAddressing)
 
+	if config.EncryptInterface != "" {
+		if err := connector.DisableRpFilter(config.EncryptInterface); err != nil {
+			log.Warn("Rpfilter could not be disabled on %s, node to node encryption may fail", config.EncryptInterface)
```
Comment (joestringer) [inline pkg/datapath/linux/datapath.go]:
```suggestion
			log.WithField(logfields.Interface, config.EncryptInterface).Warn("Rpfilter could not be disabled, node to node encryption may fail")
```

**Review by joestringer — CHANGES_REQUESTED** [review]

File: pkg/datapath/linux/datapath.go (hunk 1 above)
Comment (jrfastab) [reply pkg/datapath/linux/datapath.go]: Went ahead and folded this in thanks for the hint.

**Comment by jrfastab** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]

**Review by ianvernon — APPROVED** [review]

**Pull request merged**